frontend/src/components/PostList.jsx

```diff
@@ -1,18 +1,34 @@
 import "../styles/PostList.css";
 import Post from "./Post";
+import { httpRequest } from "../utils/utils.js";
+import { useEffect, useState } from "react";
+
+export default function PostList({ group, posts, onOpen }) {
+  const [prompt, setPrompt] = useState("");
```

Comment:
if we set group.prompt as default value of the promp use state, can we skip setting the prompt within the useEffect?
```suggestion
  const [prompt, setPrompt] = useState(group?.prompt ?? '');
```